inspire_mitmproxy/services/base_service.py

```diff
@@ -57,10 +60,19 @@ def process_request(self, request: MITMRequest) -> MITMResponse:
             if interaction.matches_request(request):
                 response = interaction.response
                 interaction.execute_callbacks()
+                self.increment_interaction_count(interaction.name)
                 return response
 
         raise NoMatchingRecording(self.name, request)
 
+    def increment_interaction_count(self, interaction_name: str):
+        try:
+            self.interactions_replayed[interaction_name]['num_calls'] += 1
+        except KeyError:
+            if interaction_name not in self.interactions_replayed:
+                self.interactions_replayed[interaction_name] = {}
+            self.interactions_replayed[interaction_name]['num_calls'] = 1
```

Comment:
these 3 lines can be just `self.interactions_replayed.setdefault(interaction_name, {'num_calls': 1})`